numpy/f2py/cfuncs.py

```diff
@@ -548,7 +548,7 @@
 #if defined(_MSC_VER)
 #define F2PY_THREAD_LOCAL_DECL __declspec(thread)
 #elif defined(__STDC_VERSION__) \\
-      && (__STDC_VERSION__ >= 201112L) \\
+      && (__STDC_VERSION__ == 201112L) \\
       && !defined(__STDC_NO_THREADS__)
```

Comment:
```suggestion
      && !defined(__STDC_NO_THREADS__)
      // For C++ 11 only
```

Reply:
Why is this needed in C++11 but not c++14?

Reply:
Or did you mean "C11", in which case my question applies to C17

Reply:
`__STDC_NO_THREADS__` is defined only in C11, see https://en.cppreference.com/w/c/thread.
C17 and others don't have the macro constant `__STDC_NO_THREADS__` defined, IIUC.

Reply:
My reading of that page is "introduced in C11" not "only in C11"

Reply:
I realized that this is confusing stuff:
- threads.h is part of C11
- Implementations don't need to provide threads.h (see https://www.wikiwand.com/en/C_standard_library about threads.h being conditional feature), hence the need for `__STDC_NO_THREADS__`
- threads.h support was implemented in glibc 2.28 and is not available in earlier versions of (C11) glibc

@eric-wieser comment regarding C17 makes sense.

Reply:
C17 concern has been addressed: `__STDC_NO_THREADS__` existence depends on glibc version, see discussion below.